### PR TITLE
Add log specification and modify log information

### LINKED ISSUE #59: Server error when the client disconnects abnormally.
8f54c1c99e48cfd18e96344c73214d7
Server will close the connection if the client disconnects. After disconnection, it should not throw an error but just give an info about the disconnection.

### PR DESCRIPTION
**Pull request by mond77**
close #59

### CODE REVIEW

**Comment by mond77** [comment]
env_logger can not record log to file. Does it matter?

**Comment by uran0sH** [comment]
> env_logger can not record log to file. Does it matter?

We also can provide another logger to log files at the same time.

**Comment by uran0sH** [comment]
Why close #59?

File: src/rpc/server.rs [hunk 1]
```diff
@@ -74,24 +74,29 @@ pub async fn receive<H: Handler + std::marker::Sync + std::marker::Send + 'stati
...
         {
-            debug!("parse_request, start");
+            let id = connection.id();
+            debug!("{} parse_request, start", id);
             let header = match connection.receive_request_header(&mut read_stream).await {
                 Ok(header) => header,
                 Err(e) => {
-                    debug!("parse_request, header error: {}", e);
+                    if e.to_string() == "early eof" {
+                        warn!("connection {} is closed abnormally.", id);
```
Comment (mond77) [inline src/rpc/server.rs:83]: #59 bug fixed here

File: src/rpc/connection.rs [hunk 2]
```diff
@@ -322,13 +322,15 @@ impl ClientConnectionAsync {
 }
 
 pub struct ServerConnection {
+    id: u32,
```
Comment (luanshaotong) [inline src/rpc/connection.rs:325]: Change to name string may be better.

File: src/rpc/server.rs (hunk 1 above)
Comment (uran0sH) [reply src/rpc/server.rs:83]: It's better to split two PRs. And printing a message isn't enough.

File: src/rpc/connection.rs (hunk 2 above)
Comment (mond77) [reply src/rpc/connection.rs:325]: server side is passively connected, what kind of name will be proper

**Review by luanshaotong — APPROVED** [review]
LGTM

**Pull request merged**